### PR TITLE
[Meta]: Welcome contributions and add code coverage requirement

### LINKED ISSUE #154: Increase code coverage with unit/integration tests
Now that we have SonarCloud enabled, we will probably want to increase [our code coverage](https://sonarcloud.io/component_measures?id=ad3154_ISO11783-CAN-Stack&metric=coverage&view=list) percentage because it will likely bring up the reputation of the stack.

What's needed to close this issue:
- [x] #155 
- [x] #157
- [x] #162  
- [ ] Add a note in the CONTRIBUTING.md about test coverage

Though I don't have much experience with automated tests, so any feedback is welcome

### PR DESCRIPTION
**Pull request by GwnDaan**
Adds a requirement to the `CONTRIBUTING.md` file regarding the test coverage. Also tried to increase the welcomingness of the first few sentences of the file. Inspiration: https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines

Closes #154

### CODE REVIEW

**Review by ad3154 — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,7 +27,8 @@ Our style rules and PR reviews are based loosely on Autosar's `Guidelines for th
 * Doxygen should compile with no warnings when run at the root of the project with the command `doxygen doxyfile`
 * Absolutely no code shall be added that is under a more strict license than MIT or which has not had conditions met to be distributed under our license
 * Builds must pass the compilation github action
-* C++14 or earlier is required to help provide the maximum compatibility to all "modern" compilers
+* C++14 or newer is required to help provide the maximum compatibility to all "modern" compilers
```
Comment (ad3154) [inline CONTRIBUTING.md:30]: Oh, one thing, when I specified `earlier` I did mean `older` not newer. The idea was to maintain compatibility with embedded compilers, many of which support 14 at the newest. For example ARM Compiler 6

Comment (ad3154) [reply CONTRIBUTING.md:30]: image

Comment (GwnDaan) [reply CONTRIBUTING.md:30]: Ohh that totally makes sense! I'll change it back

Comment (GwnDaan) [reply CONTRIBUTING.md:30]: Maybe it's more clear to state?:
```suggestion
* Only features present in C++14 can be used to help provide the maximum compatibility to all "modern" compilers
```

Comment (ad3154) [reply CONTRIBUTING.md:30]: That also works!

**Review by ad3154 — APPROVED** [review]

**Review by ad3154 — APPROVED** [review]

**Pull request merged**